source/player/Player.hx

```diff
@@ -53,8 +53,9 @@ class Player extends FlxSprite {
 
 	override function kill() {
 		reset(FlxG.width / 2, FlxG.height / 2);
-		health = maxHealth;
+		// health = maxHealth; //This line resets the players health everytime kill() is called?
 		currentPower.inUse = false;
 		currentPower.usable = true;
+		health--; // Update the players lives or health
 	}
```

Comment:
With how the game functions now, custom behavior on the player's `kill()` method is basically unnecessary, right? 

If you wanted, I guess `kill()` could be the trigger that kicks off the game over sequence in the Playstate (you could take the code that's inside the `if (player.health <= 0) {` check and make it so `player.kill()` makes it happen instead). If you're not doing that, though, I think you can fully get rid of this method override and let default `kill()` just do its thing.